src/components/share-structure/EditShareStructure.vue

```diff
@@ -240,11 +240,11 @@ export default class EditShareStructure extends Mixins(CurrencyLookupMixin) {
         (v: string) => !(this.shareClasses
           .find((s, index) => {
             // Don't apply uniqueness check to self
-            return index !== this.activeIndex && s.name.split(' Shares')[0].toLowerCase() === v.toLowerCase()
+            return index !== this.activeIndex && s.name?.split(' Shares')[0].toLowerCase() === v.toLowerCase()
           })
         ) || 'Class name must be unique')
       rules.push(
-        (v: string) => !(v.split(' ').some(r => this.excludedWordsListForClass.includes(r.toLowerCase()))) ||
+        (v: string) => !(v && v.split(' ').some(r => this.excludedWordsListForClass.includes(r.toLowerCase()))) ||
```

Comment:
Doesn't the rule on line 239 already ensure that v is truthy?

Reply:
That's what I thought as well, but put this in here to get rid of the error - tried in all places where split occured

Reply:
Yeah I needed to have the v check.. I think it validates all of the rules and doesn't short circuit

Reply:
I believe it evaluates all rules until it finds a failing one (returns string instead of True)...?
https://vuetifyjs.com/en/components/forms/#rules

Better to fix the unit test though.

Reply:
I'm curious what `v` is on line 247 (if it's not falsy).

(If this is burning up your time then I'm OK with this code change.)

Reply:
`        this.shareStructure.name = name.substr(0, name.indexOf(' Shares'))` 
probably not including Shares in the name

Reply:
Hard to tell, debugging breakpoints aren't showing the correct line with this project. I'm just leaving it for now.